### PR TITLE
[#46] 판매 조회

### PR DESCRIPTION
**Pull request by JinDDung2**
## 작업 요약

- 사용자의 판매 조회

## 작업 내용

> (체크 박스를 모두 선택해야 merge가 활성화 됩니다.)

- [x] 총 판매 금액 조회
- [x] 내 판매 상품 중 사용 확정 된 판매 내역들 조회

### CODE REVIEW

**Comment by JinDDung2** [comment]
### 📝 테스트 커버리지 리포트입니다
|File|Coverage [97.6%]|:green_apple:|
|:-|:-:|:-:|
|[PageSize.java](https://github.com/f-lab-edu/givemeticon/blob/9aa555d1bfddf7d2c0ff14e4163a24b709f2b154/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fcommon%2Futils%2Fconstants%2FPageSize.java)|100%|:green_apple:|
|[SaleItemTradeFacade.java](https://github.com/f-lab-edu/givemeticon/blob/9aa555d1bfddf7d2c0ff14e4163a24b709f2b154/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleItemTradeFacade.java)|100%|:green_apple:|
|[SaleTradeFacade.java](https://github.com/f-lab-edu/givemeticon/blob/9aa555d1bfddf7d2c0ff14e4163a24b709f2b154/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleTradeFacade.java)|100%|:green_apple:|
|[SaleController.java](https://github.com/f-lab-edu/givemeticon/blob/9aa555d1bfddf7d2c0ff14e4163a24b709f2b154/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fcontroller%2FSaleController.java)|100%|:green_apple:|
|[TradeService.java](https://github.com/f-lab-edu/givemeticon/blob/9aa555d1bfddf7d2c0ff14e4163a24b709f2b154/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Ftrade%2Fservice%2FTradeService.java)|98.77%|:green_apple:|
|[SaleService.java](https://github.com/f-lab-edu/givemeticon/blob/9aa555d1bfddf7d2c0ff14e4163a24b709f2b154/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fservice%2FSaleService.java)|92.92%|:green_apple:|

|Total Project Coverage|94.06%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/jinddung2/givemeticon/domain/sale/controller/SaleController.java [hunk 1]
```diff
@@ -24,6 +28,8 @@ public class SaleController {
     private final SaleCreationFacade saleCreationFacade;
     private final SaleService saleService;
     private final SaleItemFacade saleItemFacade;
+    private final SaleTradeFacade saleTradeFacade;
+    private final SaleItemTradeFacade saleItemTradeFacade;
```
Comment (f-lab-kyle) [inline src/main/java/com/jinddung2/givemeticon/domain/sale/controller/SaleController.java:32]: 아직 큰 문제는 안보이지만 의존이 많아지면 컨트롤러를 분리하는것도 좋아보입니다.

File: src/main/java/com/jinddung2/givemeticon/domain/sale/controller/dto/MySaleDto.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import java.time.LocalDate;
+import java.util.Date;
+
+@Getter
+@NoArgsConstructor
+public class MySaleDto {
+
+    private String itemName;
+    private LocalDate expirationDate;
+    private Date isBoughtDate;
```
Comment (f-lab-kyle) [inline src/main/java/com/jinddung2/givemeticon/domain/sale/controller/dto/MySaleDto.java]: 둘중에 하나로 맟추는게 좋을것 같아요.

File: src/main/java/com/jinddung2/givemeticon/domain/sale/service/SaleService.java [hunk 3]
```diff
@@ -72,4 +76,13 @@ public List<SaleDto> getAvailableSalesForItem(Item item) {
                 .map(SaleDto::of)
                 .collect(Collectors.toList());
     }
+
+    public List<Sale> getMySales(int userId) {
+        return saleMapper.findMySalesBySellerId(userId);
+    }
+
+    public List<Sale> getMySales(int userId, int page) {
+        Map<String, Object> pageInfo = makePagingParamMap(userId, page, SALE.getSize());
```
Comment (f-lab-kyle) [inline src/main/java/com/jinddung2/givemeticon/domain/sale/service/SaleService.java:85]: size도 파라미터로 받고 바깥에서 SALE.getSize()를 넘겨야 할지 고민이 되긴 하네요.

File: src/test/java/com/jinddung2/givemeticon/domain/sale/service/SaleServiceTest.java [hunk 4]
```diff
@@ -136,6 +138,36 @@ void get_Sales_By_ItemId_Success() {
         List<SaleDto> result = saleService.getAvailableSalesForItem(item);
 
         Assertions.assertEquals(3, result.size());
+    }
+
+    @Test
+    @DisplayName("내 판매목록을 조회한다.")
```
Comment (f-lab-kyle) [inline src/test/java/com/jinddung2/givemeticon/domain/sale/service/SaleServiceTest.java]: 테스트 이름은 가급적 Assert 기준으로 작성하는게 좋아보입니다.

**Comment by JinDDung2** [comment]
### 📝 테스트 커버리지 리포트입니다
|File|Coverage [97.6%]|:green_apple:|
|:-|:-:|:-:|
|[PageSize.java](https://github.com/f-lab-edu/givemeticon/blob/c3a15beea23245ef6ebb05d3d069b3bb4b205972/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fcommon%2Futils%2Fconstants%2FPageSize.java)|100%|:green_apple:|
|[SaleItemTradeFacade.java](https://github.com/f-lab-edu/givemeticon/blob/c3a15beea23245ef6ebb05d3d069b3bb4b205972/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleItemTradeFacade.java)|100%|:green_apple:|
|[SaleTradeFacade.java](https://github.com/f-lab-edu/givemeticon/blob/c3a15beea23245ef6ebb05d3d069b3bb4b205972/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleTradeFacade.java)|100%|:green_apple:|
|[SaleController.java](https://github.com/f-lab-edu/givemeticon/blob/c3a15beea23245ef6ebb05d3d069b3bb4b205972/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fcontroller%2FSaleController.java)|100%|:green_apple:|
|[TradeService.java](https://github.com/f-lab-edu/givemeticon/blob/c3a15beea23245ef6ebb05d3d069b3bb4b205972/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Ftrade%2Fservice%2FTradeService.java)|98.77%|:green_apple:|
|[SaleService.java](https://github.com/f-lab-edu/givemeticon/blob/c3a15beea23245ef6ebb05d3d069b3bb4b205972/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fservice%2FSaleService.java)|92.92%|:green_apple:|

|Total Project Coverage|94.06%|:green_apple:|
|:-|:-:|:-:|

**Comment by JinDDung2** [comment]
### 📝 테스트 커버리지 리포트입니다
|File|Coverage [97.6%]|:green_apple:|
|:-|:-:|:-:|
|[PageSize.java](https://github.com/f-lab-edu/givemeticon/blob/82a5f60a71fe90361095692ab6b5dda7838e4666/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fcommon%2Futils%2Fconstants%2FPageSize.java)|100%|:green_apple:|
|[SaleItemTradeFacade.java](https://github.com/f-lab-edu/givemeticon/blob/82a5f60a71fe90361095692ab6b5dda7838e4666/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleItemTradeFacade.java)|100%|:green_apple:|
|[SaleTradeFacade.java](https://github.com/f-lab-edu/givemeticon/blob/82a5f60a71fe90361095692ab6b5dda7838e4666/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Ffacade%2FSaleTradeFacade.java)|100%|:green_apple:|
|[SaleController.java](https://github.com/f-lab-edu/givemeticon/blob/82a5f60a71fe90361095692ab6b5dda7838e4666/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fcontroller%2FSaleController.java)|100%|:green_apple:|
|[TradeService.java](https://github.com/f-lab-edu/givemeticon/blob/82a5f60a71fe90361095692ab6b5dda7838e4666/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Ftrade%2Fservice%2FTradeService.java)|98.77%|:green_apple:|
|[SaleService.java](https://github.com/f-lab-edu/givemeticon/blob/82a5f60a71fe90361095692ab6b5dda7838e4666/src%2Fmain%2Fjava%2Fcom%2Fjinddung2%2Fgivemeticon%2Fdomain%2Fsale%2Fservice%2FSaleService.java)|92.92%|:green_apple:|

|Total Project Coverage|94.06%|:green_apple:|
|:-|:-:|:-:|

**Pull request merged**